### PR TITLE
Add GetConstraints() to gRPC service.

### PR DESCRIPTION
**Pull request by cschuet**

### CODE REVIEW

File: cartographer/mapping/pose_graph.cc [hunk 1]
```diff
@@ -131,24 +149,7 @@ proto::PoseGraph PoseGraph::ToProto() {
   }
```
Comment (gaschler) [inline cartographer/mapping/pose_graph.cc]: drive by: reserve constraints().size()?

File: cartographer_grpc/handlers/get_constraints_handler.h [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    : public framework::RpcHandler<google::protobuf::Empty,
+                                   proto::GetConstraintsResponse> {
+ public:
+  void OnRequest(const google::protobuf::Empty& request) override {
+    auto constraints = GetContext<MapBuilderServer::MapBuilderContext>()
+                           ->map_builder()
+                           .pose_graph()
+                           ->constraints();
+    auto response =
+        cartographer::common::make_unique<proto::GetConstraintsResponse>();
```
Comment (gaschler) [inline cartographer_grpc/handlers/get_constraints_handler.h]: same here: reserve?

File: cartographer/mapping/pose_graph.cc (hunk 1 above)
Comment (cschuet) [reply cartographer/mapping/pose_graph.cc]: Done.

File: cartographer_grpc/handlers/get_constraints_handler.h (hunk 2 above)
Comment (cschuet) [reply cartographer_grpc/handlers/get_constraints_handler.h]: Done.

**Comment by cschuet** [comment]
@wally-the-cartographer merge

**Comment by wally-the-cartographer** [comment]
Merge requested by authorized user cschuet. Merge queue now has a length of 1.

**Pull request merged**